mpas_analysis/shared/generalized_reader/generalized_reader.py

```diff
@@ -124,9 +126,31 @@ def open_multifile_dataset(fileNames, calendar, simulationStartTime=None,
                                  startDate=startDate,
                                  endDate=endDate)
 
-    ds = xarray.open_mfdataset(fileNames,
-                               preprocess=preprocess_partial,
-                               decode_times=False, concat_dim='Time', **kwargs)
+    kwargs = {'decode_times': False,
+              'concat_dim': 'Time'}
+
+    softLimit = resource.getrlimit(resource.RLIMIT_NOFILE)[0]
+
+    # use autoclose if we will use more than half the soft limit on open files
+    autoclose = len(fileNames) > softLimit/2
```

Comment:
Because we are now doing task-parallelism `autocloseSF=2` may be way too low.  Intuitively, for a single process I would have guessed `autocloseSF=3` would be sufficient.  

How should we handle this here?  Maybe it is better to specify `autoclose = len(fileNames) > softLimit/(autocloseSF*num_tasks)`, where I'm assuming we can easily get the number of tasks `num_tasks` here.  If so, `autocloseSF=2` is probably fine.